symbolic_pymc/tensorflow/printing.py

```diff
@@ -41,11 +47,50 @@ def println(self, obj):
         self.buffer.write(self.format(obj) + "\n")
         self.buffer.flush()
 
+    def depth_check(self):
+        if isinstance(self.depth_index, (list, tuple, range)):
+            return self.depth_count in self.depth_index
+        elif self.depth_index is None:
+            return True
+        else:
+            raise ValueError("Depth needs to be a tuple, list or range type.")
+
 
-def tf_dprint(obj, printer=None):
+def tf_dprint(obj, printer=None, depth_index=None):
     """Print a textual representation of a TF graph.
 
     The output roughly follows the format of `theano.printing.debugprint`.
+
+    with graph_mode():
+        t1 = [[[1, 2], [2, 3]], [[4, 4], [5, 3]]]
+        t2 = [[[7, 4], [8, 4]], [[2, 10], [15, 11]]]
+        t3 = tf.concat([t1, t2], -1)
+
+    In [1]: tf_dprint(t3)
```

Comment:
Use standard Python console formatting (i.e. `>>>`).  I believe there's an IPython magic for that.